lib/faktory_worker/worker.ex

```diff
@@ -140,21 +152,21 @@ defmodule FaktoryWorker.Worker do
   end
 
   defp handle_fetch_response({{:ok, job}, conn}, state) do
-    job_supervisor =
-      state.worker_config
-      |> faktory_name()
-      |> FaktoryWorker.JobSupervisor.format_supervisor_name()
+    job_supervisor = job_supervisor_name(state)
 
     job_ref =
       Task.Supervisor.async_nolink(
         job_supervisor,
         state.worker_module,
         :perform,
         job["args"],
-        # make this whatever is set for the 'reserve_for' minus 20 seconds
-        shutdown: :infinity
+        shutdown: :brutal_kill
```

Comment:
🔨

Reply:
I originally misread what this value is doing. It's the time that the task is allowed to run for after telling it to stop, not the maximum time to let the task run. `:brutal_kill` makes sure it stops when we say so.